doc/support_matrix.yml

```diff
@@ -0,0 +1,130 @@
+---
+
+# Defines the different support statuses.
+support_keys:
+  supported:
+    icon: ✅
+    description: supported
+  supported_with_caveats:
+    icon: ⚠️
+    description: supported with some caveats
+  needs_testing:
+    icon: ❔
+    description: needs testing; support may be present but we do not currently test it
+  being_reworked:
+    icon: ♻️
+    description: rework in progress, expect breaking changes
+  wip:
+    icon: 🚧
+    description: work in progress
+  not_currently_supported:
+    icon: ❌
+    description: not currently supported by RIOT-rs
+  not_available:
+    icon: 🚫
+    description: not available on this piece of hardware
+
+# Defines existing pieces of functionality, hardware or software.
+functionalities:
+  - name: gpio
+    title: GPIO
+    description:
+  - name: debug_output
+    title: Debug Output
+    description:
+  - name: logging
+    title: Logging
+    description:
+  - name: user_usb
+    title: User USB
+    description:
+  - name: wifi
+    title: Wi-Fi
+    description:
+  - name: ethernet_over_usb
+    title: Ethernet over USB
```

Comment:
How is this a property of the hardware? I'd expect this to be a functional property of the USB stack available, and possibly of some scalar hardware parameters like available non-control endpoints. Or do we just not have those details available yet, so "Ethernet over USB" is a baked version of those parameters?

Reply:
I'm having some issues on stm32h7, where usb-serial works fine but usb-ethernet does not. Not sure how fine-grained we want this matrix, but just because something should work doesn't mean it actually does, and that reality can become very fine grained.